### ISSUE TITLE
Wrong channel ordering in space_to_depth 

### ISSUE THREAD

**Issue by trungnt13** [tensorflow/tensorflow#53291]
In the example provided for `space_to_depth` function here: https://www.tensorflow.org/api_docs/python/tf/nn/space_to_depth

If an input tensor has a larger depth

```python
x = [[[[1, 2, 3], [4, 5, 6]],
      [[7, 8, 9], [10, 11, 12]]]]
```
Then, `space_to_depth` output is: `[ 1  2  3  4  5  6  7  8  9 10 11 12]`, slice along channel dimension (i.e., `x[0, 0, 0, :]`)

While Pytorch `pixel_unshuffle` return: `[ 1  4  7 10  2  5  8 11  3  6  9 12]`

The issue is the inconsistent in Tensorflow's behavior that is when the number of channels is _1_

```python
x= [[[[1], [4]],
       [[7], [10]]]]
```

then, the output is `[ 1  4  7 10]` which is the same as Pytorch. It is expected that when the number of channels is increasing, it should apply the `space_to_depth` to each channel separately, then concatenate the outputs along channel dimension.

However, the way Tensorflow does space_to_depth in the case of multiple channels is that it does select the elements along channel-axis which is wrong.

---

Another example, with the full code, if the input is RGGB Bayer Raw image, then TensorFlow output would be in the wrong order in case the number of channels is greater than 1.

```python
# pip install pytorch==1.10.0
# pip install tensorflow==2.6.1
# pip install einops==0.3.2
import os

os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'

import torch
from torch.nn.functional import pixel_unshuffle
import tensorflow as tf
import numpy as np
from einops import rearrange, repeat

rggb = np.asarray([[1, 2],
                   [2, 3]])

# === Only 1 channel
image = rearrange(rggb, 'h w -> h w 1')
image = repeat(image, 'h w c -> (2 h) (2 w) (1 c)')
print('RGGB image:', image.shape)
print(image[:, :, 0])

img1 = tf.nn.space_to_depth(
    rearrange(image, 'h w c -> 1 h w c'),
    2).numpy()
img2 = pixel_unshuffle(
    rearrange(torch.from_numpy(image), 'h w c -> 1 c h w'),
    2).numpy()

print('Tensorflow:', img1.shape)
print('Pytorch:', img2.shape)

img1 = rearrange(img1, '1 h w c -> h w c')
img2 = rearrange(img2, '1 c h w -> h w c')

print('Tensoflow:')
print(img1[0, 0, :])

print('Pytorch:')
print(img2[0, 0, :])

print('Match:', np.allclose(img1, img2))


# === Repeat to have 2 channels
print()
image = rearrange(rggb, 'h w -> h w 1')
image = repeat(image, 'h w c -> (2 h) (2 w) (2 c)')
print('RGGB image:', image.shape)
print(image[:, :, 0])

img1 = tf.nn.space_to_depth(
    rearrange(image, 'h w c -> 1 h w c'),
    2).numpy()
img2 = pixel_unshuffle(
    rearrange(torch.from_numpy(image), 'h w c -> 1 c h w'),
    2).numpy()

print('Tensorflow:', img1.shape)
print('Pytorch:', img2.shape)

img1 = rearrange(img1, '1 h w c -> h w c')
img2 = rearrange(img2, '1 c h w -> h w c')

print('Tensoflow:')
print(img1[0, 0, :])

print('Pytorch:')
print(img2[0, 0, :])

print('Match:', np.allclose(img1, img2))
```

output

```
RGGB image: (4, 4, 1)
[[1 2 1 2]
 [2 3 2 3]
 [1 2 1 2]
 [2 3 2 3]]
Tensorflow: (1, 2, 2, 4)
Pytorch: (1, 4, 2, 2)
Tensoflow:
[1 2 2 3]
Pytorch:
[1 2 2 3]
Match: True

RGGB image: (4, 4, 2)
[[1 2 1 2]
 [2 3 2 3]
 [1 2 1 2]
 [2 3 2 3]]
Tensorflow: (1, 2, 2, 8)
Pytorch: (1, 8, 2, 2)
Tensoflow:
[1 1 2 2 2 2 3 3]
Pytorch:
[1 2 2 3 1 2 2 3]
Match: False
```

**Comment by mohantym** [comment]
Hi @Saduf2019 ! Could you please look at this issue! Attaching Gist in [2.6](https://colab.research.google.com/gist/mohantym/c8ae0482ec0fad0e6d4c6f84702a8c29/github_53291.ipynb#scrollTo=fesXfhtJEtHx),[2.7](https://colab.research.google.com/gist/mohantym/b524955337f79e4df87cdc1620e3be9a/github_53291.ipynb#scrollTo=7lUUAnumE11k) and [nightly ](https://colab.research.google.com/gist/mohantym/97c088f37bc56bba94f7253c34a4d908/github_53291.ipynb#scrollTo=fesXfhtJEtHx)for reference. Thanks!

**Comment by sachinprasadhs** [comment]
Hi, did you try with different `data_format`, the default one is channel last("NHWC"), you can try channel first("NCHW") to see if it is matching.
